### ISSUE TITLE
Blank screen appears when opening popup in Firefox

### ISSUE THREAD

**Issue by belgacoin** [MetaMask/metamask-extension#9196]
**[Edited by MetaMask] If you are experiencing this issue, before reinstalling MetaMask, [read this on how you can help us out by providing us with more information](https://github.com/MetaMask/metamask-extension/issues/9196#issuecomment-1034044323). Thank you!**

---

Hello,
Browser details:
 - MacOS 10.14.6 (Mojave) on Macbook Air (2014)
 - Browser: Firefox 79.0
 - MetaMask Version: 8.0.7 (last updated: August 12, 2020)

So, I guess after its update, my Metamask wallet went blank.
![image](https://user-images.githubusercontent.com/4160284/90051911-b53de300-dcd8-11ea-836b-1840310c2bc1.png)

Apparently, back when I wrote down the seed words, there must have been some mistake because when restoring the wallet on another computer, an unknown address with 0 funds was showed. I guess, I mixed up something I don't know.

**Comment by Gudahtt** [comment]
Thanks for the report!

How long did you wait for the popup to render? It can sometimes be slow to initialize, and unfortunately it does just show as a blank white rectangle sometimes while initialization is happening.

I did try briefly to reproduce this issue locally but could not.

**Comment by belgacoin** [comment]
Hi, I did leave it open for long enough time. It might be related to the update. Meanwhile I could restore it on Brave browser since Ive found the correct seed.

**Comment by smartins** [comment]
Same here. Restarted browser. Disabled uBlock Origin. Same "blank" window.

![image](https://user-images.githubusercontent.com/396734/90167510-a405dc80-dd93-11ea-85a6-58d89613eac1.png)

**Comment by smartins** [comment]
Update: I've removed Metamask from FireFox and re-installed it. Seems to have fixed the blank issue.

**Comment by ouillepouille** [comment]
same issue. 
scary.

**Comment by tschubotz** [comment]
I'm experiencing this issue as well. Blank white screen, no way to use Metamask.

Firefox version: 82.0.2
Metamask version: 8.1.2
OS: MacOS 10.15.6

**Comment by jonramster** [comment]
Me, too.

Firefox/Windows

https://whatismybrowser.com/w/AEA6P9N

**Comment by Bibi10** [comment]
Even uinstall and install didn't solve it @metamaskbot

**Comment by tercel** [comment]
Me too

MetaMask: 8.1.3
Firefox: 82.0.2
OS: MacOS 10.15.5

**Comment by Gudahtt** [comment]
@tschubotz @jonramster @Bibi10 @tercel
Thanks for letting us know! Could you please contact support? You can do so here: https://metamask.zendesk.com/hc/en-us/requests/new

Unfortunately we have been unable to reproduce this problem, but getting state logs should help. It would be greatly appreciated if you could contact support and submit state logs; they can walk you through the steps to get logs in this scenario.

**Comment by tschubotz** [comment]
@Gudahtt I had to already reset my Metamask since I needed it. So I assume the state logs are gone? 
I'll Metamask contact support via Zendesk in case it happens again though.

**Comment by Gudahtt** [comment]
@tschubotz Yes, that is correct, the state logs would be gone after reinstalling. Thanks anyway!

**Comment by Incedis** [comment]
Thought I would add my 2cents. 
Issue is fixed (at least for me) after latest update of Firefox (82.0.2-3), literally minutes ago. 
What a pain not to have acces to Metamask.

**Comment by tschubotz** [comment]
>  Thanks for letting us know! Could you please contact support? You can do so here: https://metamask.zendesk.com/hc/en-us/requests/new

@Gudahtt I just now ran into the issue again - blank Metamask. Reach out to the support link now.

**Comment by tschubotz** [comment]
@Gudahtt my support request id on the Metamask Zendesk is `68274`.

**Comment by tschubotz** [comment]
@Gudahtt your colleague suggest reinstalling the extension. I just thought that you guys wanted to take a look at the state logs?

![image](https://user-images.githubusercontent.com/1380529/101159419-b13e2400-362d-11eb-9be7-1da5e1353bc7.png)

**Comment by Gudahtt** [comment]
Thanks @tschubotz! The error you sent us was helpful. We got some information from another user as well, so I have a few new leads to investigate.

For everyone in the thread, can you remember what you were doing when this first happened? I haven't seen any reports yet that included that detail. I know these instances happened a while back so you probably don't recall, but if you do (or if you see it again), any additional context might help!

**Comment by tschubotz** [comment]
> Thanks @tschubotz! The error you sent us was helpful. We got some information from another user as well, so I have a few new leads to investigate.

@Gudahtt I actually didn't send you guys any of my logs, yet. Upon `logState(true)` I can the error `Uncaught TypeError: window.getCleanAppState is not a function`. And I have yet to hear back from your colleague about this one.

I'd like to reset my Metamask soon since having to switch browsers now constantly is taking a toll on my productivity.

> For everyone in the thread, can you remember what you were doing when this first happened?

I'm not 100% sure anymore, but I think in my case, it happened after a browser-restart, i.e. I actually wanted to log back in to my Metamask. I don't remember the last (Metamask) action I did before that browser restart though.

**Comment by Gudahtt** [comment]
>I actually didn't send you guys any of my logs, yet. Upon logState(true) I get the error Uncaught TypeError: window.getCleanAppState is not a function. And I have yet to hear back from your colleague about this one.

Yep, this is what I meant! This tells us that something failed very early in the initialization code. This is interesting, and different from a few other reports we've seen, so we may be looking at multiple distinct bugs here.

**Comment by sauris** [comment]
> Thanks @tschubotz! The error you sent us was helpful. We got some information from another user as well, so I have a few new leads to investigate.
> 
> For everyone in the thread, can you remember what you were doing when this first happened? I haven't seen any reports yet that included that detail. I know these instances happened a while back so you probably don't recall, but if you do (or if you see it again), any additional context might help!

Same issue here, my pc froze and I restarted it.
Firefox 84.0.2 (64-bit) - Ubuntu 20.04
Metamask 9.0.1
Other extensions: uBlock Origin

I got exactly the same running on Win10, what both had in common is using more than 1 account, Binance Chain.

The sad part is that as addons are synchronized, removing the addon in windows removed it in Linux as well, and to add insult to injury there is no documented way in Firefox to get the Vault data as there is for Chrome.

![image](https://user-images.githubusercontent.com/35593074/105036507-89990300-5a97-11eb-98c3-85e33b9c61dd.png)
Could you please help me? Nothing from this thread or others has helped.

**Comment by belgacoin** [comment]
>> can you remember what you were doing when this first happened?
>
> i was firing up Firefox (on Mac). Wanted to login to metamask but that window stayed just plain white.

Sent from ProtonMail Mobile

Aan di, jan. 19, 2021 om 13:49, sauris <notifications@github.com> schreef:

**Comment by michaelcohen716** [comment]
getting this now 

```
MetaMask: Content script lost connection to "MetaMask Background Multiplex". undefined contentscript.js:1:117588
MetaMask: Muxed traffic for channel "metamask-provider" failed. undefined contentscript.js:1:117501
MetaMask: Content script lost connection to "MetaMask Background Legacy Multiplex". undefined contentscript.js:1:117588
MetaMask: Muxed traffic between channels "provider" and "metamask-provider" failed. undefined contentscript.js:1:120109
MetaMask: Muxed traffic for channel "publicConfig" failed. undefined contentscript.js:1:117501
MetaMask: Lost connection to "MetaMask". inpage.js:1:17727
Error: MetaMask: Disconnected from MetaMask background. Page reload required.
```

page reload did not change anything. disabled and re-enabled...no change.

anyone have any ideas?

**Comment by burtchensu** [comment]
Hit the same error:
` MetaMask: Disconnected from MetaMask background. Page reload required.`

By steps:
* connect to metamask
* close chrome
* open chrome

Ubuntu 20, Chrome Version 88.0.4324.150 , Metamask 9.0.4

**Comment by vanjan** [comment]
I'm getting the same blank issue on FF as of few days ago. Here's what's in my console:

![Screenshot 2021-02-10 at 09 50 56](https://user-images.githubusercontent.com/130294/107486664-989a3f00-6b85-11eb-921c-308721566145.png)

**Comment by L-F-Escobar** [comment]
Im getting the same issue on FF, mac... Metamask has really just not been working on FF lately, really disappointed with it

**Comment by dskarasev** [comment]
Same issue

**Comment by gigilove2001** [comment]
What can I use for an an IPhone ??
Fishing apps keep appearing on Home Screen .
Thank U

**Comment by jmahhh** [comment]
Another Firefox update, another blank screen. 

```

86.0
Firefox Release

February 23, 2021
```

**Comment by jmahhh** [comment]
> 
> 
> > I actually didn't send you guys any of my logs, yet. Upon logState(true) I get the error Uncaught TypeError: window.getCleanAppState is not a function. And I have yet to hear back from your colleague about this one.
> 
> Yep, this is what I meant! This tells us that something failed very early in the initialization code. This is interesting, and different from a few other reports we've seen, so we may be looking at multiple distinct bugs here.

any progress on this? Seems to happen around updates for me, see #9756

**Comment by NinjaTurtle007** [comment]
> Another Firefox update, another blank screen.
> 
> ```
> 
> 86.0
> Firefox Release
> 
> February 23, 2021
> ```

same here, just after updating to firefox 86 on linux
the extension also got updated today btw

**Comment by jmahhh** [comment]
#4894 am I correct in seeing this behaviour has occurred since 2018? How can an issue that effectively renders someone's Metamask inaccessible drag on for this long? @Gudahtt

**Comment by skyminert** [comment]
Is there any solution for this floating around anywhere?  It's quite a significant issue, no?

**Comment by NinjaTurtle007** [comment]
> Is there any solution for this floating around anywhere? It's quite a significant issue, no?

It is significant issue, and no there isn't a solution......
there seems to be bounty for this bug, but so far no solution....

**Comment by dskarasev** [comment]
I noticed that this happens when I try to login via Metamask, before I unlocked it with my wallet password.
When you click the Metamask button on the site, the plugin window does not appear and after that the module constantly shows a white window and without reinstalling the plugin, further work is not possible.
I hope this helps.

**Comment by cb1629966** [comment]
Having this same issue on Firefox. Opened a New Issue to complain and the dev deleted it because it was already covered here, but none of the fixes I am seeing have fixed it for me. 

I have tried closing the browser multiple times, manually going to the Menu and Exiting, restarting the computer 3x, disabling the extension and readding it, and still getting a blank screen. Could someone PLEASE help me here?

**Comment by L-F-Escobar** [comment]
Yea I can help you, use brave

**Comment by cb1629966** [comment]
@L-F-Escobar I am using brave on mobile, but MUCH prefer Firefox on the laptop.... is this the only solution for this issue? I certainly hope not...

**Comment by Jofairden** [comment]
Is this still an issue? I keep getting blank screens on my metamask!!! Any way to solve this?

**Comment by blascokoa** [comment]
Same issue here, 2 or 3 days ago it stopped working, Any way to solve it without delete it?

**Comment by mwaddip** [comment]
> Unfortunately we have been unable to reproduce this problem, but getting state logs should help. It would be greatly appreciated if you could contact support and submit state logs; they can walk you through the steps to get logs in this scenario.

To reproduce this you need to make the browser crash, or turn off your computer by switching the power. It does not happen consistently, but the only times I encountered the blank screen, it's after my computer crashed.

**Comment by L-F-Escobar** [comment]
I fixed this by switching to brave, tired of dealing with this issue

**Comment by khodarahmi** [comment]
Same issue on

Ubuntu 20.04
Firefox 86.0 (64-bit)

![image](https://user-images.githubusercontent.com/4487716/119527229-61971480-bd95-11eb-8244-e592a77c91f2.png)

**Comment by domi1981** [comment]
i've got the same issue. my ticket is 183630
I think it has to do with the firefox update. Hopefully next firefox update will fix this as someone mentioned in this threat.
I think the next release is within a week

**Comment by kewpiedoll** [comment]
Same issue, FF v88.0.1 on macOS Mojave 10.14.5. Occurred after a computer crash and hard reboot (I was interacting with metamask with and a website at the time).

Unable to use metamask currently (blank white screen). Probably will need to completely recover, based on what I'm reading in this thread. Might be time to reinstall and try Brave?
<img width="1225" alt="Screen Shot 2021-05-27 at 9 37 20 AM" src="https://user-images.githubusercontent.com/1570301/119864560-97401880-becf-11eb-8196-66bc870881c1.png">

**Comment by will-holley** [comment]
Having encountered this recently (M1 crashed with Firefox open), re-installing fixed it (until a subsequent crash).

**Comment by kewpiedoll** [comment]
> Having encountered this recently (M1 crashed with Firefox open), re-installing fixed it (until a subsequent crash).

Are you going to try Brave browser?

**Comment by will-holley** [comment]
> > Having encountered this recently (M1 crashed with Firefox open), re-installing fixed it (until a subsequent crash).
> 
> Are you going to try Brave browser?

No. Firefox is my primary dev environment.

**Comment by greg-o-ry** [comment]
New update, new blank screen :/

**Comment by JoeDon16** [comment]
How is this not fixed yet? So ridiculous

**Comment by othella31** [comment]
blank screen on firefox today..

**Comment by tschubotz** [comment]
Same for me, Metamask just turned blank today again. 
Firefox: 88.0
MacOS: 11.2.3

**Comment by DomoCom** [comment]
Same for me. Metamask is blank, i didnt backup my seed phrase. Can someone please help?

**Comment by marbella-web** [comment]
I have it blank also 
Firefox now on version 89.0.1 
I backed up some seed phrases but I have quite a few wallets I did not have backed up because I thought I could rely on the software. What could cause an over not display html code, unless there some kind of JS error some where. 

Note the strange blue icon if I right click and hit select all 
![IMG_20210621_202547](https://user-images.githubusercontent.com/56161121/122817491-e0696980-d2cf-11eb-94a4-c0bbca39c405.jpg)

![IMG_20210621_200120](https://user-images.githubusercontent.com/56161121/122817575-fc6d0b00-d2cf-11eb-801a-6db63b88ec68.jpg)


  "version": "9.6.1",
  "applications": {
    "gecko": {
      "id": "webextension@metamask.io",
      "strict_min_version": "68.0"

Get this error when clicking the toolbar icon 

![image](https://user-images.githubusercontent.com/56161121/122818704-70f47980-d2d1-11eb-8a8a-31ac20eca71b.png)

**Comment by TimDaub** [comment]
Same problem here mac os catalina 10.15.7 and ff v90

**Comment by bos1234** [comment]
what's the fix for this issue? It's been pending for quite some time now with quite a few users running into the same problem.

**Comment by brazrei** [comment]
Same problem here! Metamask Blank on firefox 89.0.2, ubuntu 18

**Comment by wnz99** [comment]
Having the same problem:

- MetaMask 9.7.1
- Ubuntu 20.04
- Firefox 89.0.2 (64-bit)

I can see the following error in the addon debug:

![image](https://user-images.githubusercontent.com/12447140/125672778-f26b85ca-4cf4-4bfb-8af4-809d47fed069.png)

**Comment by ChosunOne** [comment]
Just experienced significant financial loss from this issue.  I tried all the methods above and learned that my seed phrase I had backed up did not recover the same address.  I reinstalled the extension but now the previous account I had in here is nowhere to be found.

**Comment by NinjaTurtle007** [comment]
> Just experienced significant financial loss from this issue. I tried all the methods above and learned that my seed phrase I had backed up did not recover the same address. I reinstalled the extension but now the previous account I had in here is nowhere to be found.

Just keep creating new accounts/addresses until you get to the one you used previously.
All accounts are derrived from the same seed phrase, it just recreates just the first one upon restoring it.

**Comment by talentlessguy** [comment]
Blank screen on Firefox 91

![image](https://user-images.githubusercontent.com/35937217/129550543-2a5960f3-111b-4421-9da6-745464ea00a3.png)

**Comment by dpazdan** [comment]
maybe worth checking this:
https://support.mozilla.org/en-US/questions/1195232

Start Firefox in Safe Mode to check if one of the extensions ("3-bar" menu button or Tools -> Add-ons -> Extensions) or if hardware acceleration is causing the problem.

    switch to the DEFAULT theme: "3-bar" menu button or Tools -> Add-ons -> Appearance
    do NOT click the "Refresh Firefox" button on the Safe Mode start window 

    https://support.mozilla.org/en-US/kb/troubleshoot-firefox-issues-using-safe-mode
    https://support.mozilla.org/en-US/kb/troubleshoot-extensions-themes-to-fix-problems

**Comment by z3r0kira** [comment]
![metamask debug 2](https://user-images.githubusercontent.com/68660419/132624906-6be0678b-5e4c-4daf-873e-ef9e6d50d015.JPG)
![metamask debug 3](https://user-images.githubusercontent.com/68660419/132624909-6ec48dbe-5c42-4d7d-bbef-9c2b9922866d.JPG)
![metamask debug1](https://user-images.githubusercontent.com/68660419/132624911-2da46f06-d7aa-4f6e-a3a0-bd0665ef6397.JPG)

same issues. not sure if these help. started happening a few days ago thought it was an update related thing. hope it fixes itself cus i dont seem to have pulled a backup for metamask and am just now realizing i have a decent chunk tied to it

**Comment by enderphan94** [comment]
MacOS BigSur 11.6
Firefox 92.0.1
Metamask 10.1.0
<img width="397" alt="Screen Shot 2021-10-05 at 12 37 10 PM" src="https://user-images.githubusercontent.com/15033155/135966405-396c247c-9a75-4065-a1d8-52c439b7e484.png">

**Comment by smeshny** [comment]
MacOS BigSur 11.6
Firefox 93.0
Metamask 10.1.0

After Firefox 93.0 update Metamask not working.

Showing blank window. Any advice?

**Comment by KaiRo-at** [comment]
Sorry but that latest comment is just BS. The versions that people post on are mostly release versions and just because Firefox is a more serious and secure browser when it comes to clean API usage doesn't make it worse then Google's one (which is known to discard their users' privacy and instead spy on users and use their data for ad targeting).

**Comment by mehrdadb3** [comment]
why u didnt fix this problem its for 3 times i get blank screen on metamask in firefox.every time i have to reinstal metamask

**Comment by ilya-afanasev** [comment]
I am using firefox 93.0 (64 bits) and I get blank screen on metamask too. How to fix?

**Comment by shimmyshine** [comment]
MetaMask is near unusable.  I've had this happen a handful of times, end up having to reinstall and reload everything, adding all of my chains over again.  It's an easy 20 minute process.

I've noticed it happens mostly when firefox is unexpectedly closed (system update, power outage, etc...).  I haven't restored it this go around yet, and of course have backed up my seed phrase & private keys.  So I can provide any logs necessary if I don't need it before then.  But this issue is unsettling to someone who's never had it happen before and annoying to someone who it's happened to multiple times.

**Comment by enderphan94** [comment]
> it happens mostly when firefox is unexpectedly closed

exactly!

**Comment by barrybaz** [comment]
Is the error cause known yet? From what I've seen, it could be related to the password login: The issue occured for me when I had the browser (firefox) store the metamask password and perform the login automatically. When I don't have this, i.e. when I manually enter the metamask password, the issue doesn't seem to occur. Perhaps that's helpful?

**Comment by phpandrew** [comment]
I just encountered this today... I'm shocked to see this issue has been open for a year now.

**Comment by boredland** [comment]
I have this since I enabled Hardware acceleration in FF

**Comment by cmackeen** [comment]
Same issue with Firefox 94.0 (64-bit) on ubuntu, was working a couple of days ago. Never used firefox sync or whatever, and had no issues with this extension so far. Seems like an important bug.

**Comment by donzthefonz** [comment]
Just started experiencing this today too... looks like Metamask auto updated today. Uninstall and reinstall does not fix the issue. 
Running POP OS (Ubuntu base) and Firefox 94.0
Debug logs:
![image](https://user-images.githubusercontent.com/62137152/145073633-2bf37746-4fb1-4785-a8a3-e61eb8d6bbcf.png)

**Comment by dazito** [comment]
Same issue here. Every week I get this bug and I am forced to reinstall metamask on firefox.
I see this bug has been happening since (at least) 2018 and there are several bug reports filled but I see zero or almost zero action from Metamask.

I know Metamask is open source but when you charge a fee on your swap features and therefore make (a lot of) money from it I expect you to show, at least, some consideration to your users.

Every week I have to configure all different networks and import all different tokens I have on my several accounts. Besides tedious this is very time consuming.

**Comment by znmcmhn** [comment]
This just started happening to me 2 hours ago.  
My street had a Power outtage late last night that shut my PC down .....and my ISP had a 2-hour Outtage/Maintenance period this morning
And then when I booted up my PC ....Microsoft windows performed a update

And now I'm stuck here/confused on what to do and cant find my seed phrase too test/reinstall Meta mask ? 
This is not a good place to be when we have zero-access at all to even open Metamask.
It's just pure blank/ Non-functional.

**Comment by joshuavial** [comment]
I have this issue too - happened last week when my computer (Mac) shut down suddenly. On restart metamask is white screen. Have logged issue with support but got some garbage reply about elastic tokens.

**Comment by linuxh3lper** [comment]
Hi @znmcmhn contact me on my email linuxiscool@yandex.com I'll help you find your seed :) it's not easy... but I'll write you a guide.

**Comment by will-holley** [comment]
@znmcmhn – beware @linuxh3lper; he has no Github history and there's a high probability will try to steal your funds.

**Comment by linuxh3lper** [comment]
@will-holley You humorist :D I just want to help him. I had this problem... @znmcmhn good guide is also here https://medium.com/coinmonks/lost-metamask-wallet-the-forensic-way-c1871c3768f3 but it's not easy for Windows users.

**Comment by mudbile** [comment]
I just experienced this, and it was only after reading the accounts of others did I realise I too had just gone through an unexpected restart- my computer crashed. So I think that's a pretty solid lead. 

Same errors as @Subarasheese [above](https://github.com/MetaMask/metamask-extension/issues/9196#issuecomment-936542141). 

I went and deleted everything from the extensions folder for my Mozilla profile, started Firefox, realised I couldn't be arsed reinstalling everything and that it would be better to find the MetaMask file/s specifically, undid the mass-delete and all my extensions were back (I just had to re-enable them) except for MetaMask. Reinstalled MetaMask, imported seed phrase and she's good to go.

So long story short, there's something in the extensions folder screwing things for MM after an unexpected shutdown sometimes (my computer's crashed before without this issue). Deleting all the files in there and reinstalling everything is probably a simple solution.

**Comment by joshuavial** [comment]
I had another unexpected restart and it did the same thing - the coinmonks medium article above was exactly what I needed to easily recover the data.

**Comment by cmdBee** [comment]
I found a simple solution (Mac) that I didn't see mentioned anywhere else. This solution should hopefully work if your MetaMask is blank after your computer crashed. Here are the steps:

# Solution:

## 1. Find your local metamask extension folder

Should be here:

`/Users/{your-username}/Library/Firefox/Profiles/{your-profile}/storage/default/moz-extension+++{extension-uuid}^userContextId=XXXXXXX`

`{your-username}` is your computer username. `{your-profile}` is your Firefox profile. To find your `extension-uuid` paste `about:debugging#addons` into your address bar and scroll down until you see MetaMask. The UUID will be labelled as `Internal UUID`.

## 2. Locate the file whose name is a number

Inside the extension folder there will be another folder called `idb` and inside that folder is another folder named something like `3647222921wleabcEoxlt-eengsairo.files`. Finally, inside that folder is a file whose name is a number (`1` or `2` or even `3020`). Mine was called `3039` and was ~300kb in size.

## 3. Make a copy of this file

Copy this file and increment the name by 1. So in my case I named the copy `3040`. If your file is `58` name the copy `59`.

## 4. Restart Firefox

Fully restart Firefox (press the `command` and `Q` keys together to properly quit the application).

## 5. Open MetaMask

Now just open MetaMask and it should be working again. If not, go back to step 3 and increment the copied file name again and restart Firefox again. In my case I would rename my copied file from `3040` to `3041`.

# Explanation

So it turns out that every time you open Firefox this file's name is incremented by anywhere between 1 to 10 (the next time I opened Firefox my file was renamed from `3040` to `3048`). This file name is stored in the `.sqlite` file found in the parent folder of the file and if your computer crashes these might get out of sync. In my case the file was named `3039` but in the `.sqlite` file it was referenced as `3040`.

I'm not an extension developer so I am not sure if this is a Firefox quirk or if it's a MetaMask quirk.

I hope this helps anybody facing this issue!

**Comment by mackwage** [comment]
I can vouch for @cmdBee 's solution.. except in my case the number ended up being off but over a thousand. So wanted to add an additional tip for others who run into this issue. 

As @cmdBee stated, the needed ID is referenced in the associated sqlite database. So if you navigate to the parent directory (idb), you should see the associated sqlite file (note the sqlite file extension). You can easily open and browse the data using SQLite Browser (https://sqlitebrowser.org/). 

Once you open the sqlite file, click the 'Browse Data' tab and in the table 'file', you will see a column named 'id'. Whatever number is in the below row in this column is the ID number of the file it is looking for so you should rename your file to whatever this number is.

**Comment by znmcmhn** [comment]
Just a friendly note for those confused and having trouble. I'm laughing because I ....succeeded!!!!!!!!!! : ) 
There appear to be several similar routes/methods so don't give up!
If you still have the extension on your browser your account(s) are completely recoverable!!
I had no experience whatsoever but kept pushing for a long confusing 5-7 days but still got through it.

I know it is an anxious long task but just keep pushing through!! Don't let your doubts get to you and be careful!

If anyone would like directional guidance I can help cut through the time via telegram.
It would be fun to help someone as a buddy/friend combatant/and I'd enjoy a second go...you'll find it relatively easy with not being alone with encouragement. Take care!

**Comment by znmcmhn** [comment]
Just a further note:
My Metamask faulted due too a sudden power outtage/ rendering the Metamask Ui dead.
- Thus, ....killing "any" Metamask re-installation ..too the browser!!!!!

So, A new browser installation were required too!
(-Two copies of the same browser) 

But don't dare install a new browser until your Metamask seed is entirely recovered entirely in hand!!

**Comment by dbadol** [comment]
Thanks to @cmdBee and @mackwage !
The process did work perfectly on Windows 10 / Firefox 96.0.2. Thank you so much!
Interestingly enough, I had to rename the file to a _lower_ number!...

**Comment by hilvmason** [comment]
[https://app.zenhub.com/files/42009758/485c9615-6742-4034-8c90-87a9992218bf/download](https://app.zenhub.com/files/42009758/485c9615-6742-4034-8c90-87a9992218bf/download)

**Comment by Pyanne** [comment]
Unfortunately @cmdBee 's solution didn't solve the problem for me. The path to the file was this:

` /Users/<username>/Library/Application Support/Firefox/Profiles/y0kflvyl.default-release/storage/default/moz-extension+++{extension-uuid}/idb/3619119340leogcaarlof.files/5`

Changing the number from 5 to 6,7,8,9 or 4 did not help so I went the @mackwage way. Opening the .sqlite file in SQLite Browser I saw the number should be 5 ! There were nothing wrong.

I am still looking for a solution.

**Comment by znmcmhn** [comment]
I think this solution is alot more user friendly compared to the 2 others I had seen. The demo-pictures help immensely. It worked for me.
https://community.metamask.io/t/metamask-blank-screen-on-firefox-but-no-seed/12971/24

**Comment by stoplion** [comment]
Still happening in 2022

**Comment by Jason-Black** [comment]
Feb 6 2022 - Issue is still happening.  You might want to try to get this fixed before you spend ANY time on any other features - this is completely game breaking for me.  My seed phrase is hours away from me in a secure location - this is the 4th freaking time I have had to completely uninstall and restore.  It's obviously a UI or catch issue bc people report this after a power outage.  Please for the love of god put a team on this - for someone who didn't have their seed phrase backed up, this would render their money (the most important thing in our lives) inaccessible.  It puts people in an absolutely desperate situation, you have to get on this ASAP.  PLEASE.

**Comment by Antmann11** [comment]
Made an account here to say that @cmdBee solution with the followup from @mackwage worked perfectly for me! THANK YOU!

**Comment by jpuri** [comment]
While looking at this issue I came across a similar issue in firefox, sometimes Metamask extension is blank in firefox when open the first time - but if the user re-opens it it is not more blank. This is demoed here:

https://user-images.githubusercontent.com/2182307/153161851-df087301-5d13-48bb-9ddf-5fe44a3e42d8.mov

I found that reason for this issue is that as as extension is loaded a connection stream for RPC calls is created between the extension UI and the background. At appears that at times this has issue and write over the stream fail [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/metaRPCClientFactory.js#L84). Thus [callback](https://github.com/MetaMask/metamask-extension/blob/develop/ui/index.js#L35) in UI initialisation is never invoked.

It is again hard to say why this happens in firefox, but this issue does not do any big hard to user account. User can just close and re-open the extension and should be able to see UI if it was the connection stream that was causing the issue.

**Comment by mcmire** [comment]
Hi everyone,

Thank you for all of your reports around this issue. We appreciate that you want MetaMask to work great for everyone.

Based on what we've heard and seen so far, there seem to be two reasons for why users could see a persistent blank screen. We are currently investigating the second and most recent, highlighted by @cmdBee, which is that an unexpected crash or shutdown can corrupt the database stored in your Firefox profile data that MetaMask uses to hold information such as your list of accounts and your settings. Unfortunately this type of bug is extremely difficult to trigger ourselves. If you encounter this issue, we would greatly appreciate it if you were to help us out by following these instructions:

1. Don't uninstall MetaMask! Doing so will remove the data we need to help you.
2. In Firefox, type in `about:debugging` in your address bar and press Enter.
3. Click on "This Firefox" in the sidebar.
4. Within the Extensions section, find MetaMask. Note the Internal UUID (a string of random letters and numbers with dashes separating them). We'll come back to this in a second.
5. Click on the "Inspect" button next to MetaMask.
6. Click on the "Console" tab if it is not already selected.
7. Right-click on any of the console messages in the space below and select Export Visible Messages To > File. This will give you a log file. Save the file somewhere easily accessible.
8. Open your Firefox profile directory by following the "How do I find my profile?" section on [this page](https://support.mozilla.org/en-US/kb/profiles-where-firefox-stores-user-data).
9. Within your profile directory you will see a "storage" folder. Click into this.
10. Click into the "default" folder.
11. Now find two folders. One is called `moz-extension++++<UUID>` and the other is called `moz-extension++++<UUID>^userContextId=<some number>` (where "\<UUID\>" is the Internal UUID for MetaMask you saw earlier). Select both of these folders and choose "Compress to ZIP File" (Windows) or "Compress" (macOS).
12. Create a ticket in our [help desk](https://metamask.zendesk.com/hc/en-us/requests/new?ticket_form_id=360001389932).
    1. Please make sure to select the version you're using. You can do this by going to "Addons & themes", then Extensions, then clicking on MetaMask and scrolling down until you see "Version".
    2. For the Subject field, please use the words "blank screen" so that we can find your issue more easily.
    3. Please upload two attachments: 
        1. The log file you created in step 7.
        2. The ZIP file that you created in step 11.

Thank you for your cooperation and your patience as we continue to look into this!

**Comment by anonqoder** [comment]
I run a build generated from  ```yarn build dev``` and could reproduce the error and got a slightly better stacktrace
The issue is many dependencies are ```.min.js```, single line ```.js``` that makes debugging difficult
Is there a way to produce a fully non minified build?

**Comment by anonqoder** [comment]
Actually the documented way of running from the development branch is not working, tried both on Firefox and Chrome

- Edit ```..metamaskrc.dist``` , put in with my Infura ID
- ```yarn setup```
- ```yarn start```
- Followed  https://github.com/MetaMask/metamask-extension/blob/develop/docs/add-to-firefox.md

Result: blank page. The extension icon is not even added to the toolbar

If you have any updated docs on how to set this up, I could provide logs and debug for my case

**Comment by wcr1pt0** [comment]
> I found a simple solution (Mac) that I didn't see mentioned anywhere else. This solution should hopefully work if your MetaMask is blank after your computer crashed. Here are the steps:
> # Solution:
> ## 1. Find your local metamask extension folder
> 
> Should be here:
> 
> `/Users/{your-username}/Library/Firefox/Profiles/{your-profile}/storage/default/moz-extension+++{extension-uuid}^userContextId=XXXXXXX`
> 
> `{your-username}` is your computer username. `{your-profile}` is your Firefox profile. To find your `extension-uuid` paste `about:debugging#addons` into your address bar and scroll down until you see MetaMask. The UUID will be labelled as `Internal UUID`.
> ## 2. Locate the file whose name is a number
> 
> Inside the extension folder there will be another folder called `idb` and inside that folder is another folder named something like `3647222921wleabcEoxlt-eengsairo.files`. Finally, inside that folder is a file whose name is a number (`1` or `2` or even `3020`). Mine was called `3039` and was ~300kb in size.
> ## 3. Make a copy of this file
> 
> Copy this file and increment the name by 1. So in my case I named the copy `3040`. If your file is `58` name the copy `59`.
> ## 4. Restart Firefox
> 
> Fully restart Firefox (press the `command` and `Q` keys together to properly quit the application).
> ## 5. Open MetaMask
> 
> Now just open MetaMask and it should be working again. If not, go back to step 3 and increment the copied file name again and restart Firefox again. In my case I would rename my copied file from `3040` to `3041`.
> # Explanation
> 
> So it turns out that every time you open Firefox this file's name is incremented by anywhere between 1 to 10 (the next time I opened Firefox my file was renamed from `3040` to `3048`). This file name is stored in the `.sqlite` file found in the parent folder of the file and if your computer crashes these might get out of sync. In my case the file was named `3039` but in the `.sqlite` file it was referenced as `3040`.
> 
> I'm not an extension developer so I am not sure if this is a Firefox quirk or if it's a MetaMask quirk.
> 
> I hope this helps anybody facing this issue!

Hello,

I tried but it didnt works. The orginal file name was: 71070 -  I renamed it to 71071 - 71072 - 71073 ... 71080 but unfortunately the problem was not fixed. Could you please advice? Thanks in advance and sorry to bother man.

**Comment by wcr1pt0** [comment]
> **[Edited by MetaMask] If you are experiencing this issue, before reinstalling MetaMask, [read this on how you can help us out by providing us with more information](https://github.com/MetaMask/metamask-extension/issues/9196#issuecomment-1034044323). Thank you!**
> 
> Hello, Browser details:
> 
>     * MacOS 10.14.6 (Mojave) on Macbook Air (2014)
> 
>     * Browser: Firefox 79.0
> 
>     * MetaMask Version: 8.0.7 (last updated: August 12, 2020)
> 
> 
> So, I guess after its update, my Metamask wallet went blank. ![image](https://user-images.githubusercontent.com/4160284/90051911-b53de300-dcd8-11ea-836b-1840310c2bc1.png)
> 
> Apparently, back when I wrote down the seed words, there must have been some mistake because when restoring the wallet on another computer, an unknown address with 0 funds was showed. I guess, I mixed up something I don't know.

Hello, did you fix it? I'm having the same problem...

**Comment by anonqoder** [comment]
The only reliable fix for this issue is to rollback to Metamask 0.13.0 and disable all updates until this get fixed

1. Grab from mozilla add-ons page https://addons.mozilla.org/firefox/downloads/file/3860271/metamask-10.3.0-an+fx.xpi and install it
2. Go to ```about:addons``` in the browser
3. Look for Metamask. In the ```Allow automatic updates``` option, check ```Off```

**Comment by rossgalloway** [comment]
> I can vouch for @cmdBee 's solution.. except in my case the number ended up being off but over a thousand. So wanted to add an additional tip for others who run into this issue.
> 
> As @cmdBee stated, the needed ID is referenced in the associated sqlite database. So if you navigate to the parent directory (idb), you should see the associated sqlite file (note the sqlite file extension). You can easily open and browse the data using SQLite Browser (https://sqlitebrowser.org/).
> 
> Once you open the sqlite file, click the 'Browse Data' tab and in the table 'file', you will see a column named 'id'. Whatever number is in the below row in this column is the ID number of the file it is looking for so you should rename your file to whatever this number is.

This solution worked beautifully. Unlike the other post saying that the number needed to be incremented higher, my id number was lower than the file name by 2 digits. I replaced with the correct number and its all working fine again.

Anyway, I submitted the logs to the helpdesk. Kudos to @cmdBee for the tip on using the SQlite browser!

**Comment by wcr1pt0** [comment]
> > I can vouch for @cmdBee 's solution.. except in my case the number ended up being off but over a thousand. So wanted to add an additional tip for others who run into this issue.
> > As @cmdBee stated, the needed ID is referenced in the associated sqlite database. So if you navigate to the parent directory (idb), you should see the associated sqlite file (note the sqlite file extension). You can easily open and browse the data using SQLite Browser (https://sqlitebrowser.org/).
> > Once you open the sqlite file, click the 'Browse Data' tab and in the table 'file', you will see a column named 'id'. Whatever number is in the below row in this column is the ID number of the file it is looking for so you should rename your file to whatever this number is.
> 
> This solution worked beautifully. Unlike the other post saying that the number needed to be incremented higher, my id number was lower than the file name by 2 digits. I replaced with the correct number and its all working fine again.
> 
> Anyway, I submitted the logs to the helpdesk. Kudos to @cmdBee for the tip on using the SQlite browser!

Hey, thanks for replying :)

I tried but it doesnt work...

The original file name was 71070 I renamed it to 71071/9... no luck. Then I tried SQlite browser

Renamed the file to 71069 but still not working...

<img width="613" alt="Screen Shot 2022-03-15 at 16 27 27" src="https://user-images.githubusercontent.com/78519867/158456139-f4ae8bed-3937-4afa-b5ea-2839f27885a6.png">

**Comment by miguelmota** [comment]
The [solution](https://github.com/MetaMask/metamask-extension/issues/9196#issuecomment-1012084136) by @cmdBee worked perfectly! h/t

For my case the number actually needed to be rolled back and I was able to see what it was by searching the sqlite file. 

Also if you're on linux the paths will be under `~/.mozilla` folder.

**Comment by stfl** [comment]
My numbered file is an empty folder, or the folder is invisible to me..
It might be related to snap..

```
file /home/stefan/.mozilla/firefox/32qlgygq.default-release/storage/default/moz-extension+*${UUID}/idb/*.files/[0-9]*

/home/stefan/.mozilla/firefox/32qlgygq.default-release/storage/default/moz-extension+++4fc1e9a8-29e0-4431-8641-bcf066423ad3/idb/3619119340leogcaarlof.files/1: snappy framed data
```

**Comment by esaumell** [comment]
> > I can vouch for @cmdBee 's solution.. except in my case the number ended up being off but over a thousand. So wanted to add an additional tip for others who run into this issue.
> > As @cmdBee stated, the needed ID is referenced in the associated sqlite database. So if you navigate to the parent directory (idb), you should see the associated sqlite file (note the sqlite file extension). You can easily open and browse the data using SQLite Browser (https://sqlitebrowser.org/).
> > Once you open the sqlite file, click the 'Browse Data' tab and in the table 'file', you will see a column named 'id'. Whatever number is in the below row in this column is the ID number of the file it is looking for so you should rename your file to whatever this number is.
> 
> This solution worked beautifully. Unlike the other post saying that the number needed to be incremented higher, my id number was lower than the file name by 2 digits. I replaced with the correct number and its all working fine again.
> 
> Anyway, I submitted the logs to the helpdesk. Kudos to @cmdBee for the tip on using the SQlite browser!

I agree. My number needed to be decreased. Neatest solution. Thanks @cmdBee

**Comment by malawadd** [comment]
> Im currently have 3 id numbers in sqlite file id # 2,3,4. The same files exist in my idb folder .files. The largest one is 1.4mb which is #4. Do I keep the latest and delete old ones from my sqlite and .files. I think my account is confused in accessing the right profile. Does any user have multiple numbers or just 1?
> 
> Using mac /
> 
> Users//Library/Application Support/Firefox/Profiles/y0kflvyl.default-release/storage/default/moz-extension+++{extension-uuid}/idb/3619119340leogcaarlof.files
> 
> Any assistance will be appreciated

i have 3 id numbers as well , did you manage to fix it ?

**Comment by volley86** [comment]
is work

**Comment by EUA** [comment]
FYI: I found that this is issue also happening due unstable (highly overclocked) memory

**Comment by DDDDDanica** [comment]
Document by @mcmire that concludes this issue:
https://docs.google.com/document/d/1VbzJEdx1quH9s1msYrK0k5jAZ0kJ4NuL1-9Ma2N7HOk

**Comment by AlexHerman1** [comment]
this is not stale.

**Comment by gettty** [comment]
Hi folks,  

- Windows 10 
- Chrome 115.0.5790.110
- Metamask 10.34.0

![image](https://github.com/MetaMask/metamask-extension/assets/6483775/f1dc9bce-aa93-47ea-b254-15c224d9580b)
![image](https://github.com/MetaMask/metamask-extension/assets/6483775/d3fd002c-c360-4c84-a57a-f11374195ad9)

No errors in the network logs.

**Comment by AlexHerman1** [comment]
@gettty  i think that that issue may be unrelated. we're working on researching and creating a new GH issue for it.

**Comment by attila-lendvai** [comment]
same issue here... :/

filename is in sync with the id in the sqlite db.

going back to 10.3.0 only changes the failure mode: blank/white window instead of an error message.

**Comment by crispoltd** [comment]
Same issue here

Ubuntu 20.05
Firefox 117.0
10.34.0

![image](https://github.com/MetaMask/metamask-extension/assets/25205334/0c94eabd-4054-4bc0-a55d-b42e5feeed32)

**Comment by gauthierpetetin** [comment]
Hi everyone, has this bug still been experienced recently by someone?

**Comment by gauthierpetetin** [comment]
Probably related to https://github.com/MetaMask/metamask-extension/issues/22659

**Comment by SKYBITDev3** [comment]
Here is the solution for Windows Firefox users that worked for me at least a few times over the years.

Download and install [SQLite Browser](https://sqlitebrowser.org).

In Firefox:
Navigate to `about:debugging#/runtime/this-firefox`.
Scroll to MetaMask.
Copy the Internal UUID to the clipboard.

Close Firefox.

In File manager:
Navigate to `C:\Users, [your username], AppData\Roaming\Mozilla\Firefox\Profiles`
Search for the Internal UUID
From the results, navigate into the directory: `[letters and numbers].default-release\storage\default\moz-extension+++[Internal UUID]^userContextId=[numbers]\idb`
You should see a file having the name `[letters and numbers]-eengsairo.sqlite`
Copy the full path to the clipboard. 

In SQLite Browser:
Click "Open Database".
Paste the the clipboard (the full path to the SQLite file).
Right-click on the `file` table, click "Browse Table".
Copy the number in the `id` column to clipboard.

Close SQLite Browser.

In File manager:
From the idb directory navigate into `[letters and numbers]-eengsairo.files`
You should see a file whose name is a number.
Proceed to rename it, pasting the clipboard.

Start Firefox.
Load MetaMask extension.
The login page should now appear.

Thank you to @cmdBee and @mackwage for the useful tips in 2022 that still work up to today.

**Comment by gauthierpetetin** [comment]
cc @MetaMask/extension-platform for visibility

**Comment by bigr247** [comment]
@Skybitdev3      not going to lie mate but i'am NOT tech savvy.  a video tutorial of everything you just described would be a blessing.    you know more like an over the shoulder youtube type of love.  i have no doubt in my tiny mind that if i was to follow every step that you just described above.... I would probably contact life on mars for sure!   that's how bad i would screw s@#t up.  A video please :)

**Comment by Bhlynd** [comment]
> Here is the solution for Windows Firefox users that worked for me at least a few times over the years.
> 
> Download and install [SQLite Browser](https://sqlitebrowser.org).
> 
> In Firefox: Navigate to `about:debugging#/runtime/this-firefox`. Scroll to MetaMask. Copy the Internal UUID to the clipboard.
> 
> Close Firefox.
> 
> In File manager: Navigate to `C:\Users`, [your username], `AppData\Roaming\Mozilla\Firefox\Profiles` Run a file and folder search for the Internal UUID. From the results, navigate into the directory:
> 
> ```
> [letters and numbers].default-release\storage\default\moz-extension+++[Internal UUID]^userContextId=[numbers]\idb
> ```
> 
> You should see a file having the name `[letters and numbers]-eengsairo.sqlite`. Copy the full path to the clipboard.
> 
> In SQLite Browser: Click "Open Database". Paste the the clipboard (the full path to the SQLite file). Right-click on the `file` table, click "Browse Table". Copy the number in the `id` column to clipboard.
> 
> Close SQLite Browser.
> 
> In File manager: From the idb directory navigate into `[letters and numbers]-eengsairo.files`. You should see a file whose name is a number. Proceed to rename it, pasting the clipboard.
> 
> Start Firefox. Load MetaMask extension. The login page should now appear.
> 
> Thank you to @cmdBee and @mackwage for the useful tips in 2022 that still work up to today.

Followed these steps with success. Can confirm still valid today 8/05/2024.

**Comment by alpertungax1** [comment]
Perfect. Solved my issues.

**Comment by cryptodigi** [comment]
what about chrome?

**Comment by karenbollozos** [comment]
Hi, is there a solution for Mac Firefox users?

**Comment by dj-markwOrm** [comment]
I found a video tutorial to fix this problem.
[](https://www.youtube.com/watch?v=WHimZr8jsYc)

**Comment by simonindelicate** [comment]
Thanks @Bhlynd and @cmdBee and @mackwage - worked for me. 

This happened after a BSOD for me - is the issue that the extension gets out of sync with the browser when you hit a crash?

**Comment by JeaneLG** [comment]
> I can vouch for [@cmdBee](https://github.com/cmdBee) 's solution.. except in my case the number ended up being off but over a thousand. So wanted to add an additional tip for others who run into this issue.
> As [@cmdBee](https://github.com/cmdBee) stated, the needed ID is referenced in the associated sqlite database. So if you navigate to the parent directory (idb), you should see the associated sqlite file (note the sqlite file extension). You can easily open and browse the data using SQLite Browser (https://sqlitebrowser.org/).
> Once you open the sqlite file, click the 'Browse Data' tab and in the table 'file', you will see a column named 'id'. Whatever number is in the below row in this column is the ID number of the file it is looking for so you should rename your file to whatever this number is.

also from my side many thanks. I had to **decrease** (not increase) my number by one.

**Comment by souvikmishra** [comment]
Hey, I was also facing this issue, but upon following the steps I found out that I had more files that the author of the solution had while they were debugging, for eg, 

I have 3 `.sqlite` files, and 2 `.files` folders (one of which is empty) in the `/idb` directory. 
Also in the `.files` folders (the one which is not empty) I have 2 numbered files (instead of 1) and a `journals` folder (which is empty). 

Now I don't really know which numbered file i should update using which sqlite table's id. 

If anyone here is still actively monitoring this or has an idea on how it could be resolved then that would be of great help.

**Comment by davidmurdoch** [comment]
@szekelyisz https://support.metamask.io/

"Contact support" opens this window where you can ask for help (from AI or just ask for a human):

![Image](https://github.com/user-attachments/assets/a01dfc5e-25ff-4e48-998d-6e77ff4892c7)

**Comment by szekelyisz** [comment]
> @szekelyisz https://support.metamask.io/
> 
> "Contact support" opens this window where you can ask for help (from AI or just ask for a human):
> 

Thanks. An extension must have blocked that popup for me, I'll look into it.

Is this the right way to open a support ticket so I can attach the logs and the zipped files later?

**Comment by davidmurdoch** [comment]
@szekelyisz yes. https://support.metamask.io/ is the place for that.

**Comment by gauthierpetetin** [comment]
Mitigated by: https://github.com/MetaMask/metamask-extension/pull/32006

**Comment by jcalv13** [comment]
Since you have changed your architecture this trick no longer works which is a massive shame because it was useful for multiple start up issues. I am currently having issues when opening the app that it is stuck in infinite loading as mentioned here by another ( https://github.com/MetaMask/metamask-extension/issues/33586 ) but can no longer use this fix.
